src/core/ack_tracker.c

```diff
@@ -196,11 +196,12 @@ QuicAckTrackerAckPacket(
         Connection->Settings.MaxAckDelayMs == 0 ||
         (Tracker->AckElicitingPacketsToAcknowledge >= (uint16_t)Connection->PacketTolerance) ||
         (!Connection->State.IgnoreReordering &&
+         Connection->ReorderingThreshold > 0 &&
          (NewLargestPacketNumber &&
           QuicRangeSize(&Tracker->PacketNumbersToAck) > 1 && // There are more than two ranges, i.e. a gap somewhere.
             QuicRangeGet(
             &Tracker->PacketNumbersToAck,
-          QuicRangeSize(&Tracker->PacketNumbersToAck) - 1)->Count == 1))) { // The gap is right before the last packet number.
+          QuicRangeSize(&Tracker->PacketNumbersToAck) - 1)->Count == Connection->ReorderingThreshold))) {
```

Comment:
I'm thinking the logic is not actually enough (at least in some edge cases). Currently, it's saying 'if there is more than one ACK range, once the last ACK range count equals the reordering threshold, immediately send an ACK'. But if we have sporadic loss, and each ACK range never hits the reordering threshold, we won't send an ACK.

For example, if the reordering threshold is 3 packets, but we never get more than 2 in a row, then this fails. We would only look at the last range and see count is 2, which is less than 3.

I think we need to make a new helper function, `QuicAckTrackerDidHitReorderingThreshold`, which handles the case where there are multiple gaps. I _think_ the function just needs to calculate the difference between the last number in the first range, and the last number in the last range. I still would like to put this in a helper function because this logic is so complex here.